src/preferences/colorpaletteeditormodel.h

```diff
@@ -17,7 +20,7 @@ class ColorPaletteEditorModel : public QStandardItemModel {
     bool setData(const QModelIndex& index, const QVariant& value, int role = Qt::EditRole) override;
 
     void setColor(int row, const QColor& color);
-    void appendRow(const QColor& color, int hotcueIndex = kNoHotcueIndex);
+    void appendRow(const QColor& color, const QList<int>& hotcueIndicies);
```

Comment:
but how should the `HotcueIndexListItem` handle it then? Should it take the QList as a pointer as well or how am I supposed to avoid the creation of the temporary object there?

Reply:
According to the Standard c++ guideline a function that takes a raw pointer promises not to store it. 
In this case HotcueIndexListItem can copy the object it points to.

Reply:
Does this advice still apply?

Reply:
No, it is outdated.